### DISCUSSION TITLE
How To Debug Flash Apps on Packaged Browser

### DISCUSSION THREAD

**Discussion by yishayw** [airsdk/Adobe-Runtime-Support#1499] [Q&A]
Hi,

I know this is not an AIR question, but perhaps someone can point me in the right direction nonetheless. I am trying to connect fdb to a client's packaged browser. I tried simply calling run and launching the browser but no success. Any ideas?

Thanks
Yishay

**Comment by ajwfrost** [comment]
Hi @yishayw  - the normal Flash Player binaries don't support debugger functionality. You would need to request a special debugger build in order to get this working..  Can you contact the pbsupport@harman.com team with your company details and ask for a package with the debugger version of the Flash Player?
thanks

**Comment by yishayw** [comment]
Sure, I’ll get our client to contact Harman support with all the details. I was actually given a dll for my windows machine which was supposed to contain the debug fp and replaced the old one with it, so I might have already done that. Do you know off-hand if replacing the dll should have been enough? If the problem is not that I don’t have the debugger build, then it might be that the application was built for release. I will need to check that.

Thanks.

**Reply by ajwfrost** [reply]
If you have a debugger DLL loaded, then it should automatically try to connect to the FDB process .. I'm not 100% sure though as I'm wondering if there is some Chromium-based security/sandbox that may be preventing this... I know that Scout works, but that you have to manually create a configuration file in the appropriate location for the app in order to give the Flash Player the requested telemetry information. And thinking about it, I'm 99% sure that we've had customers use the DLL (i.e. Pepper Flash Player) with the debugger e.g. from Flash Builder...

Let me know if you've still got a problem, I can sync up with the support team on it..

thanks

**Comment by yishayw** [comment]
I’m waiting to see if the app is built for debug (will probably know after holidays) before investigating further. I’ll report here about findings in case others have the same issue. Thanks for the feedback.

**Comment by yishayw** [comment]
Hi Andrew,

FYI I was able to get this working. The only caveat is that I need to select ‘debug’ from the context menu. If I try to just ‘run’ in fdb and wait for the application to run and connect then we get failures with messages about missing capabilities to the player.

So flow that does work for me:


  1.  Run application
  2.  Start fdb
  3.  In fdb type ‘run’
  4.  In application select ‘debug’ from context menu.
  5.  Result: fdb connects successfully

Flow that does not work for me:


  1.  Start fdb
  2.  In fdb type ‘run’
  3.  Run application
  4.  Result: message saying player does not support features required by the debugger

I don’t have the message (d) in front of me (I’m on a different pc), but I can look for it if you don’t know what I’m talking about and need my help to debug this.

Thanks,
Yishay

From: Yishay ***@***.***>
Sent: Thursday, December 30, 2021 1:54 PM
To: ***@***.***>; ***@***.***>
Cc: ***@***.***>
Subject: RE: [airsdk/Adobe-Runtime-Support] How To Debug Flash Apps on Packaged Browser (Discussion #1499)

I’m waiting to see if the app is built for debug (will probably know after holidays) before investigating further. I’ll report here about findings in case others have the same issue. Thanks for the feedback.
